server/util/status/status.go

```diff
@@ -206,9 +222,29 @@ func UnauthenticatedErrorf(format string, a ...interface{}) error {
 	return UnauthenticatedError(fmt.Sprintf(format, a...))
 }
 
-// Wrap adds additional context to an error, preserving the underlying status code.
+// WrapError prepends additional context to an error description, preserving the
+// underlying status code and error details.
 func WrapError(err error, msg string) error {
-	return makeStatusError(status.Code(err), fmt.Sprintf("%s: %s", msg, Message(err)))
+	s, ok := status.FromError(err)
```

Comment:
status.FromError always return a valid status, even when ok is false. 
When ok is false the error is returned is an "Unknown" error.
For best compatibility, seems like it would be best not to fail on `!ok`?

Reply:
Good catch - fixed